### PR TITLE
Upgrade to newer Awaitility.

### PR DESCRIPTION
**Pull request by AmlingPalantir**
**Goals (and why)**:

**Implementation Description (bullets)**:

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2429)

### CODE REVIEW

File: atlasdb-config/src/test/java/com/palantir/atlasdb/factory/TransactionManagersTest.java [hunk 1]
```diff
@@ -146,7 +146,7 @@
     public void setup() throws JsonProcessingException {
         // Change code to run synchronously, but with a timeout in case something's gone horribly wrong
         originalAsyncMethod = TransactionManagers.runAsync;
-        TransactionManagers.runAsync = task -> Awaitility.await().atMost(2, TimeUnit.SECONDS).until(task);
+        TransactionManagers.runAsync = task -> Awaitility.await().atMost(2, TimeUnit.SECONDS).untilAsserted(task::run);
```
Comment (hsaraogi) [inline atlasdb-config/src/test/java/com/palantir/atlasdb/factory/TransactionManagersTest.java]: `task` is a Runnable I believe, why did we change this to `task::run`?

File: versions.props [hunk 2]
```diff
@@ -9,7 +9,7 @@ com.google.dagger:dagger = 2.0.2
 com.google.dagger:dagger-compiler = 2.0.2
 com.google.guava:* = 18.0
 com.googlecode.json-simple:json-simple = 1.1.1
-com.jayway.awaitility:awaitility = 1.6.5
```
Comment (hsaraogi) [inline versions.props]: Looked through the release notes, https://github.com/awaitility/awaitility/wiki/ReleaseNotes30. The changes make sense. Thanks for the contribution!

**Comment by hsaraogi** [comment]
Closing in favor of #2668.

**Comment by AmlingPalantir** [comment]
Whoops, I somehow missed that there had been responses.  task is indeed Runnable, but untilAsserted demands a ThrowingRunnable.

**Pull request closed without merging**